test/jit_hooks/test_jit_hooks.cpp

```diff
@@ -0,0 +1,270 @@
+  torch::jit::Module module =
+      torch::jit::load(path_to_exported_script_module + "_" + "test_module_same_hook_repeated" + ".pt");
+  std::vector<torch::jit::IValue> inputs;
+  inputs.push_back("a");
+
+  auto output = module(inputs);
+  std::cout << "----- module output: " << output << std::endl;
+  AT_ASSERT("a_ph_ph_outermod_inner_mod_fh_fh" == output);
+}
+
+void test_module_forward_single_input(
+    const std::string& path_to_exported_script_module) {
+  torch::jit::Module module =
+      torch::jit::load(path_to_exported_script_module + "_" + "test_module_forward_single_input" + ".pt");
+  std::vector<torch::jit::IValue> inputs;
+  inputs.push_back("a");
+
+  auto output = module(inputs);
+  std::cout << "----- module output: " << output << std::endl;
+  AT_ASSERT("pre_hook_override_name_outermod_inner_mod_fh" == output);
+}
+
+void test_module_hook_return_nothing(
+    const std::string& path_to_exported_script_module) {
+  torch::jit::Module module =
+      torch::jit::load(path_to_exported_script_module + "_" + "test_module_hook_return_nothing" + ".pt");
+  std::vector<torch::jit::IValue> inputs;
+  inputs.push_back("a");
+
+  auto output = module(inputs);
+  std::cout << "----- module output: " << output << std::endl;
+  AT_ASSERT("a_outermod_inner_mod" == output);
+}
+
+
+void test_submodule_forward_single_input(
+    const std::string& path_to_exported_script_module) {
+  torch::jit::Module module =
+      torch::jit::load(path_to_exported_script_module + "_" + "test_submodule_forward_single_input" + ".pt");
+  std::vector<torch::jit::IValue> inputs;
+  inputs.push_back("a");
+
+  auto output = module(inputs);
+  std::cout << "----- module output: " << output << std::endl;
+  AT_ASSERT("pre_hook_override_name_inner_mod" == output);
+}
+
+void test_module_multiple_hooks_single_input(
+    const std::string& path_to_exported_script_module) {
+  torch::jit::Module module =
+      torch::jit::load(path_to_exported_script_module + "_" + "test_module_multiple_hooks_single_input" + ".pt");
+  std::vector<torch::jit::IValue> inputs;
+  inputs.push_back("a");
+
+  auto output = module(inputs);
+  std::cout << "----- module output: " << output << std::endl;
+  AT_ASSERT("pre_hook_override_name2_outermod_inner_mod_fh1_fh2" == output);
+}
+
+
+void test_submodule_multiple_hooks_multiple_inputs(
+    const std::string& path_to_exported_script_module) {
+  torch::jit::Module module =
+      torch::jit::load(path_to_exported_script_module + "_" + "test_submodule_multiple_hooks_multiple_inputs" + ".pt");
+  std::vector<torch::jit::IValue> inputs;
+  torch::List<std::string> list({"a"});
+  inputs.push_back(list);
+  inputs.push_back("no_pre_hook");
+
+  auto output = module(inputs);
+  std::cout << "----- module output: " << output << std::endl;
+  std::ostringstream stream;
+  stream << output;
+  std::string output_str =  stream.str();
+  AT_ASSERT("([pre_hook_override_name, inner_mod_name], pre_hook_override2_fh1_fh2)" == output_str);
```

Comment:
Could you use `IValue` comparison here instead by creating an `ivalue::Tuple` that contains the expected list and string?